crates/block-explorers/src/lib.rs

```diff
@@ -99,6 +115,16 @@ impl Client {
         Client::builder().with_api_key(api_key).chain(chain)?.build()
     }
 
+    /// Create a new client with the correct endpoint with the chain and chosen API v2 version
+    pub fn new_v2_from_env(chain: Chain) -> Result<Self> {
+        let api_key = std::env::var("ETHERSCAN_API_KEY")?;
+        Client::builder()
+            .with_api_version(EtherscanApiVersion::V2)
+            .with_api_key(api_key)
+            .chain(chain)?
+            .build()
```

Comment:
then this can just call new ad with_version(v2)

Reply:
A little confused as to where this would be called -- within the builder or skip the builder entirely and use `with_version()` rather than `new_v2_from_env`?